components/hab/src/main.rs

```diff
@@ -771,12 +771,20 @@ fn raw_parse_args() -> (Vec<OsString>, Vec<OsString>) {
 /// Check to see if the user has passed in an AUTH_TOKEN param. If not, check the
 /// HAB_AUTH_TOKEN env var. If not, check the CLI config to see if there is a default auth
 /// token set. If that's empty too, then error.
-fn auth_token_param_or_env(m: &ArgMatches) -> Result<String> {
+fn auth_token_param_or_env(ui: &mut UI, m: &ArgMatches) -> Result<String> {
     match m.value_of("AUTH_TOKEN") {
         Some(o) => Ok(o.to_string()),
         None => {
             match henv::var(AUTH_TOKEN_ENVVAR) {
-                Ok(v) => Ok(v),
+                Ok(v) => {
+                    // Temporary warning until we deprecate Github tokens completely
+                    if !v.starts_with("_") {
+                        ui.warn(
+                            "WARNING: Github tokens are being deprecated, please migrate to a Habitat Personal Access Token instead."
```

Comment:
A docs link (or maybe just straight to https://bldr.habitat.sh/#/profile) would be useful